.editorconfig

```diff
@@ -0,0 +1,23 @@
+#
+#  http://editorconfig.org
+#
+# Please have your editor honor these settings!
+#
+
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
```

Comment:
gitignore this file and remove pls

Reply:
also, looks like `package-lock.json` got checked in as well

Reply:
Both of these inclusions are intentional and in my view at least, good things:

`package-lock.json` is supposed to be checked in because it does what it says - locks packages at specific versions that we know work. This prevents a lot of headaches common in the JS community, and is a feature.

`.editorconfig` is also supposed to be checked in - for people whose editor supports it, it will provide consistent formatting, reducing incidences of 'please don't mix tabs and spaces, and for the love of god don't mix 2-space and 4-space indents'.

Reply:
I'm not outright opposed to adding these files, but we should move that change to a separate PR.